src/Main/java/Player/Player.java

```diff
@@ -1,228 +1,239 @@
 import views.GameplayFrame;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 
 /**
  * Absztrakt alaposztaly, a konkret peldanyai az Eszkimo (Eskimo) vagy a Sarkkutato (Researcher).
  */
 
-public abstract class Player extends Entity implements OutputToString{
-    private volatile int actualHealth ;
-    private volatile int actualWorkUnit ;
-    private int maxHealth ;
-    private List<Item> items = new ArrayList<Item>();
+public abstract class Player extends Entity implements OutputToString {
+    private volatile int actualHealth;
+    private volatile int actualWorkUnit;
+    private int maxHealth;
+    private List<Item> items = new ArrayList<>();
     private boolean inWater = false;
     private ClothesEquipped clothes = new NoClothesEquipped();
-    public Player(int _actualHealth,int _actualWorkUnit,int _maxHealth,Field _field){
-        actualHealth = _actualHealth;
-        actualWorkUnit = _actualWorkUnit;
-        maxHealth = _maxHealth;
-        setField(_field);
+    private boolean endTurn = false;
+
+    protected Player(int actualHealth, int actualWorkUnit, int maxHealth, Field field) {
+        this.actualHealth = actualHealth;
+        this.actualWorkUnit = actualWorkUnit;
+        this.maxHealth = maxHealth;
+        setField(field);
         Manager.AddPlayer(this);
-    };
-    public Player(int _actualHealth,int _actualWorkUnit,int _maxHealth){
-        actualHealth = _actualHealth;
-        actualWorkUnit = _actualWorkUnit;
-        maxHealth = _maxHealth;
+    }
+
+    protected Player(int actualHealth, int actualWorkUnit, int maxHealth) {
+        this.actualHealth = actualHealth;
+        this.actualWorkUnit = actualWorkUnit;
+        this.maxHealth = maxHealth;
         Manager.AddPlayer(this);
-    };
+    }
 
     public boolean isInWater() {
         return inWater;
     }
 
-    private boolean endTurn = false;
-    public void setEndTurn(boolean _endturn){endTurn = _endturn;}
+    public void setInWater(boolean inWater) {
+        this.inWater = inWater;
+        Manager.Update(this);
+    }
 
-    public int getActualWorkUnit() {
+    public void setEndTurn(boolean endTurn) {
+        this.endTurn = endTurn;
+    }
+
+    public synchronized int getActualWorkUnit() {
         return actualWorkUnit;
     }
 
-    public synchronized void  setActualWorkUnit(int actualWorkUnit) {
+    public synchronized void setActualWorkUnit(int actualWorkUnit) {
         this.actualWorkUnit = actualWorkUnit;
     }
 
-    public int getActualHealth() {
+    public synchronized int getActualHealth() {
         return actualHealth;
     }
+
+    /**
+     * Növeli a jatekos munkaegysegeit eggyel
+     */
+    public synchronized void IncreaseWorkUnit() {
+        actualWorkUnit++;
+    }
+
     /**
-     * A jatek menedzsere
+     * Csokkenti a jatekos munkaegysegeit eggyel
      */
-    private Manager manager = Manager.getInstance();
```

Comment:
Ezekkel mi történt? Ki lettek törölve, de máshol nem találom őket...

Reply:
A Manager osztály egy statikus osztály (legalábbis nincsenek nem statikus változói és minden metódusa statikus) ezért felesleges a példányt létrehozni. (Emelett nem is volt egyáltalán használva a változó)